mainloop.h

```diff
@@ -28,7 +28,7 @@ struct endpoint_entry {
     bool remove;
 };
 
-class Mainloop : public Router {
+class Mainloop : public Router, public Timeout_Manager {
```

Comment:
Mainloop should **not** inherit from Router in the first place. Don't add double inheritance.  All you need to do is to add some forward declarations in some headers